caphandler/addresspools.go

```diff
@@ -34,8 +39,7 @@ func GetAddressPoolCollection(ctx iris.Context) {
 	fabricID := ctx.Params().Get("id")
 	// get all switches which are store under that fabric
 
-	_, ok := capdata.FabricDataStore.Data[fabricID]
-	if !ok {
+	if _, err := capmodel.GetFabric(fabricID); errors.Is(err, db.ErrorKeyNotFound) {
```

Comment:
please correct the error handling

Reply:
Incorporated comments, please review again.